src/app/3d/qgs3ddebugwidget.cpp

```diff
@@ -0,0 +1,130 @@
+/***************************************************************************
+  qgs3ddebugwidget.cpp
+  --------------------------------------
+  Date                 : November 2024
+  Copyright            : (C) 2024 by Matej Bagar
+  Email                : matej dot bagar at lutraconsulting dot co dot uk
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgis.h"
+
+#include "moc_qgs3ddebugwidget.cpp"
+#include "qgs3ddebugwidget.h"
+#include "qgs3dmapcanvas.h"
+#include "qgscameracontroller.h"
+
+Qgs3DDebugWidget::Qgs3DDebugWidget( Qgs3DMapCanvas *canvas, QWidget *parent ) : QWidget( parent ), m3DMapCanvas( canvas )
```

Comment:
it is a habit in qgis code to put the initializer list on the next line, and have each member initialized on a separate line for better readability